pkg/tso/keyspace_group_manager.go

```diff
@@ -131,8 +131,10 @@ func (s *state) getKeyspaceGroupMetaWithCheck(
 	// The keyspace doesn't belong to this keyspace group, we should check if it belongs to any other
 	// keyspace groups, and return the correct keyspace group meta to the client.
 	if kgid, ok := s.keyspaceLookupTable[keyspaceID]; ok {
-		return s.ams[kgid], s.kgs[kgid], kgid,
-			genNotServedErr(errs.ErrGetAllocatorManager, keyspaceGroupID)
+		if s.ams[kgid] != nil {
+			return s.ams[kgid], s.kgs[kgid], kgid, nil
+		}
+		return nil, s.kgs[kgid], kgid, genNotServedErr(errs.ErrGetAllocatorManager, keyspaceGroupID)
```

Comment:
when s.ams[kgid] is nil, can s.kgs[kgid] be nil at the same time?

Reply:
no. s.kgs stores all keyspace groups; s.ams[kgid] only stores allocator managers of the keyspace groups assigned to this server/pod